back/app/services/xlsx_service.rb

```diff
@@ -300,6 +301,13 @@ def header_style(s)
   def convert_to_text_long_lines html
     convert_to_text(html).gsub(/\n/, ' ')
   end
+
+  # Sheet names, derived from Cause titles for example, can only be 31 characters long,
+  # and cannot contain the characters \ , / , * , ? , : , [ , ].
+  # We are being strict and removing any character that is not alphanumeric or a space.
+  def sanitize_sheetname(sheetname)
+    sheetname[0..30].gsub(/[^A-Za-z0-9 ]/, '')
```

Comment:
Slightly nitpicking, but I'd remove the special chars before fetching the 30 first characters, if the string contains many special characters, the minimum could be greatly reduced if you do this in the current order.

Reply:
Smart idea. Will do :+1: